src/welcome/WelcomePage.js

```diff
@@ -5,11 +5,51 @@ import Box from '../components/box/Box'
 import Button from '../components/button/Button'
 import ComponentLoader from '../loader/ComponentLoader.js'
 
-const WelcomePage = ({ doUpdateIpfsApiAddress, ipfsInitFailed, ipfsConnected, ipfsReady, ipfsApiAddress }) => {
+const WelcomePage = ({ doUpdateIpfsApiAddress, apiUrl, ipfsInitFailed, ipfsConnected, ipfsReady, ipfsApiAddress }) => {
   if (!ipfsInitFailed && !ipfsReady) {
     return <ComponentLoader pastDelay />
   }
 
+  const renderConnectionStatus = () => {
```

Comment:
Minor but I'd find this easier to read if `renderConnectionStatus` was another component rather than an extra render function, like `<ConnectionStatus connected sameOrigin />`